File: 프로그래머스/0/120866. 안전지대/안전지대.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board) {
    int n = board.size();
    int m = board[0].size();
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < m; ++j) {
            if (board[i][j] == 1) {
                for (int x = i - 1; x <= i + 1; ++x) {
                    for (int y = j - 1; y <= j + 1; ++y) {
                        if (x >= 0 && x < n && y >= 0 && y < m) {
                            if (board[x][y] == 0) {
                                board[x][y] = 2;
                            }
                        }
                    }
                }
            }
        }
    }
    int answer = 0;
    
    for(int i=0;i<n;++i){
        for(int j=0;j<m;++j){
            if(board[i][j] == 0){
                answer++;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    n = len(board)
    m = len(board[0])
    
    for i in range(n):
        for j in range(m):
            if board[i][j] == 1:
                for di in range(-1, 2):
                    for dj in range(-1, 2):
                        ni = i + di
                        nj = j + dj

                        if 0 <= ni < n and 0 <= nj < m:
                            if board[ni][nj] == 0:
                                board[ni][nj] = 2

    answer = 0

    for i in range(n):
        for j in range(m):
            if board[i][j] == 0:
                answer += 1
    return answer